src/script/interpreter.h

```diff
@@ -174,7 +174,34 @@ class MutableTransactionSignatureChecker : public TransactionSignatureChecker
     MutableTransactionSignatureChecker(const CMutableTransaction* txToIn, unsigned int nInIn, const CAmount& amountIn) : TransactionSignatureChecker(&txTo, nInIn, amountIn), txTo(*txToIn) {}
 };
 
-bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript& script, unsigned int flags, const BaseSignatureChecker& checker, SigVersion sigversion, ScriptError* error = NULL);
+class ScriptExecution {
+public:
+    typedef std::vector<unsigned char> StackElementType;
+    typedef std::vector<StackElementType> StackType;
+
+    const CScript& script;
+    StackType& stack;
+    unsigned int flags;
+    const BaseSignatureChecker& checker;
+    SigVersion sigversion;
+
+    CScript::const_iterator pc;
+    CScript::const_iterator pbegincodehash;
+
+    std::vector<bool> vfExec;
+    StackType altstack;
+    int nOpCount;
+
+    ScriptExecution(StackType& stack, const CScript&, unsigned int flags, const BaseSignatureChecker&, SigVersion);
+
+    bool Eval(ScriptError* error = NULL);
+};
+
+inline bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript& script, unsigned int flags, const BaseSignatureChecker& checker, SigVersion sigversion, ScriptError* error = NULL)
+{
+    return ScriptExecution(stack, script, flags, checker, sigversion).Eval(error);
```

Comment:
Move `stack` to `::Eval()` for the time being as this is an output (also remove `stack` member from class)?